lib/main.coffee

```diff
@@ -49,8 +38,8 @@ module.exports =
         filePath = textEditor.getPath()
         command = @executablePath
         ruleset = @rulesets
-        if @projectRules
-          rulesetPath = helpers.find(filePath, 'ruleset.xml')
+        if /^[a-z0-9]+\.xml$/gi.test(@rulesets)
```

Comment:
The purpose of the `projectRules ` checkbox setting was to find a file named `ruleset.xml` and if found it will override anything in the `rulesets` field.
This filename was hardcoded, if you used a different name than `ruleset.xml` then the `projectRules` setting was of no use in your project.

Instead of adding a third field for specifying a different name this PR is merging the two settings into a single field.

If you pass `cleancode,myrules.xml` the regex check will bail out and pass the options directly to phpmd not doing a filesystem lookup.

If you only pass a single option without a path this PR will make the assumption that you want to load the ruleset file if found in the current or parent folders.

`/^[a-z0-9]+\.xml$/gi.test('myrules.xml') => true`
`/^[a-z0-9]+\.xml$/gi.test('cleancode,myrules.xml') => false`
`/^[a-z0-9]+\.xml$/gi.test('myrules.xml,cleancode') => false`
`/^[a-z0-9]+\.xml$/gi.test('myrules.xml,cleancode') => false`
`/^[a-z0-9]+\.xml$/gi.test('/myproject/some/path/myrules.xml') => false`
`/^[a-z0-9]+\.xml$/gi.test('/myrules.xml') => false`
`/^[a-z0-9]+\.xml$/gi.test('myrules.xml,extrarules.xml') => false`

What this means is that if someone used the old feature  to find a `ruleset.xml` file, that someone can simply enter `ruleset.xml` in the rules field and everything will work exactly like it does today. With the added benefit of being able to customize the file to look for.

The reasoning behind merging the options is to reduce confusion. If you have to explain the UI using inline comments like "This setting will override the previous setting" it is a sign of being ambiguous/complicated. Merging them makes it clearer what's happening and how it's configured.
I came to that conclusion after first attempting to add a third field to change the name of the file you're looking for and struggling to write a field description that was short and sweet.